packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.tsx

```diff
@@ -102,15 +102,9 @@ export class SpinButton extends BaseComponent<ISpinButtonProps, ISpinButtonState
     this._inputId = getId('input');
     this._spinningByMouse = false;
 
-    if (!props.defaultValue && props.value !== undefined) {
-      this._onValidate = props.onValidate;
-      this._onIncrement = props.onIncrement;
-      this._onDecrement = props.onDecrement;
-    } else {
-      this._onValidate = this._defaultOnValidate;
-      this._onIncrement = this._defaultOnIncrement;
-      this._onDecrement = this._defaultOnDecrement;
-    }
+    this._onValidate = props.onValidate || this._defaultOnValidate;
```

Comment:
Not sure I like either the old or new code here; let me try to explain:

For a helper like this._onValidate, which has some default logic, but perhaps the user may override it, we should do something like this:

```
private _onValidate(ev: HTMLInputEventArgs) {
  const { onValidate } = this.props;

  if (onValidate) {
    onValidate(ev);
  }
  if (!ev.defaultPrevented) {
    ...do default logic.
  }
}
```

Reply:
This was also my first thought, but I didn't want to change the original code too much.